newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrExpressionSelector/index.js

```diff
@@ -52,178 +52,167 @@ type Props<T> = {|
   onClickMore?: () => void,
   id?: ?string,
 |};
-type State<T> = {|
+type SearchState<T> = {|
   searchText: string,
   searchResults: Array<T>,
 |};
 
-export default class InstructionOrExpressionSelector<
+const InstructionOrExpressionSelector = <
   T: EnumeratedInstructionOrExpressionMetadata
-> extends React.PureComponent<Props<T>, State<T>> {
-  state: State<T> = {
+>({
+  focusOnMount,
+  instructionsInfo,
+  instructionsInfoTree,
+  selectedType,
+  onChoose,
+  iconSize,
+  useSubheaders,
+  searchPlaceholderObjectName,
+  searchPlaceholderIsCondition,
+  helpPagePath,
+  style,
+  onClickMore,
+  id,
+}: Props<T>) => {
+  const searchBarRef = React.useRef<?SearchBarInterface>(null);
+  const scrollViewRef = React.useRef<?ScrollViewInterface>(null);
+  const selectedItemRef = React.useRef<?ListItemRefType>(null);
+  const [searchState, setSearchState] = React.useState<SearchState<T>>({
     searchText: '',
     searchResults: [],
-  };
-  _searchBar: ?SearchBarInterface;
-  _scrollView = React.createRef<ScrollViewInterface>();
-  _selectedItem = React.createRef<ListItemRefType>();
-  searchApi = null;
-
-  initialInstructionTypePath = findInTree(
-    this.props.instructionsInfoTree,
-    this.props.selectedType
+  });
+  const searchApi = React.useMemo(
+    () =>
+      new Fuse(instructionsInfo, {
+        ...sharedFuseConfiguration,
+        keys: [
+          { name: 'displayedName', weight: 2 },
+          { name: 'fullGroupName', weight: 1 },
+        ],
+      }),
+    [instructionsInfo]
+  );
+  const initialInstructionTypePathRef = React.useRef<?(string[])>(
+    findInTree(instructionsInfoTree, selectedType)
   );
 
-  componentDidMount() {
-    if (this._selectedItem.current && this._scrollView.current) {
-      this._scrollView.current.scrollTo(this._selectedItem.current);
```

Comment:
I removed this logic. I don't know what it was used for: How could an item be selected at the moment the component is mounted?

Reply:
I tried to add it as an effect when the instructionsInfo prop changes so it scrolls the list so that the selected item is at the top of the list.
But I felt like it was not natural and when you switch to an object that misses the instruction (the item get unselected) and switch back to another that supports the instruction, it does not scroll to the item so it feels broken.

Reply:
Good question. I don't have the answer

Reply:
Yes, I just checked. It's not handled by this component anyway